doc/apiary/v2/fiware-ngsiv2-reference.apib

```diff
@@ -29,91 +29,79 @@ An entity is represented by a a JSON object which contain the following properti
-```
-
-* `string` (or `String`): identifies text string. They are represented by JSON strings (see https://developer.mozilla.org/en-US/docs/Glossary/String).
-
-```
-"msg": "Hello world"
-```
-
-* `boolean` (or `Boolean`): identifies a logical value, either true or false. They are represented by JSON booleans (see https://developer.mozilla.org/en-US/docs/Glossary/Boolean).
-
-```
-"active": true
-```
-
-* `date` (or `Date`):  identifies dates, in ISO8601 format. They are represented by JSON strings. Operators allowed are:
-greater than, lesser than, greater or equal, lesser or equal and range.
+* `date`:  identifies dates, in ISO8601 format. These attributes can be used with the
+  greater than, lesser than, greater or equal, lesser or equal and range query operators. Eg:
 
 ```
-"timestamp": "2017-06-17T07:21:24.238Z"
+  "timestamp": {
+    "value": "2017-06-17T07:21:24.238Z",
+    "type: "date"
+  }
 ```
 
-
-* `geo:point`: identifies a geolocation. They are represented by JSON strings. An
-entity can only have one `geo:point` attribute (TBD: this is yet under discussion, one possibility is
-there can be multiple points associated to an entity but an entity can only be located on one place).
+* `geo:point`: identifies the location of the entity in geo-location queries. If an entity have several attributes
+  of this type all them define a possible location for the entity from a geo-location query point of view. (TBD:
+  check feasibility).
 
 ```
-{
-  ...
   "location": {
     "value": "41.3763726, 2.1864475,14",
     "type": "geo:point"
   }
-}
-```
-
-At creation or update time, the type has to be coherent with the value of the attribute, e.g the following
-will result in an error response:
-
-(TBD: this behaviour is under discussion, specially the case for the "myType" with JSON number for the value)
-
 ```
-{
-  "value": "23.5", -> value should be: 23.5
-  "type": "number"
-}
 
+### Canonical format description
 
-{
-  "value": 23.5,   -> value should be: "23.5"
-  "type": "string"
-}
+This is a variant of the represention format aimed at clients that needs a regular representation of the entities
+and attributes, that can be used in the operations that support the `cannonical` option.
 
-{
-  "value": "23.5", -> value should be: "23.5" (user defined types always use JSON strings as value)
-  "type": "myType"
-}
+* Entity `type` is mandatory. If the entity has no type, JSON null is used.
+* Property `attrs` is mandatory and includes a JSON object for the attributes.
+* Each attribute is described always with `value`, `type` and `metadata`. All fields are mandatory (attributes
+  without type use JSON null for that property).
+* Each metadata is described always with `value` and `type`. Both fields are mandatory (metadata
+  without type use JSON null for that property).
 
+```
 {
-  "value": "foo, bar", -> it is not correctly formated as coordinates
-  "type": "geo:point"
+  "type": "Room",
+  "id": "Boe_Idearium",
+  "attrs": {
```

Comment:
As far as I understand, the value of the `attrs` property is a JSON object which keys are the names of the attributes... Which reason would motivate the usage of an array here? 

We moved from arrays to key-maps in the past (that's was the reason because `attributeFormat` parameter was invented in v1) because some people don't like arrays. I wouldn't like to make the same "error" again ;)